fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/models/ResourceFactoryImpl.java

```diff
@@ -193,14 +194,18 @@ private FedoraResource instantiateResource(final String transactionId,
         }
     }
 
-    private void populateResourceHeaders(final FedoraResourceImpl resc, final ResourceHeaders headers,
-                                         final Instant version) {
+    private void populateResourceHeaders(final String transactionId, final FedoraResourceImpl resc,
+                                         final ResourceHeaders headers, final Instant version) {
         resc.setCreatedBy(headers.getCreatedBy());
         resc.setCreatedDate(headers.getCreatedDate());
         resc.setLastModifiedBy(headers.getLastModifiedBy());
         resc.setLastModifiedDate(headers.getLastModifiedDate());
         resc.setParentId(headers.getParent());
-        resc.setEtag(headers.getStateToken());
+        final Instant containmentUpdated = containmentIndex.containmentLastUpdated(transactionId, resc.getFedoraId());
```

Comment:
This code is only relevant when loading a container, correct? If so, I think we should only do this lookup when necessary.

This code would also be a good candidate for caching in the future.

Reply:
I'm guessing that there will be similar code to this for membership? It might be nice to abstract the etag computation out into its own class, but you don't necessarily need to do so here.

Reply:
One of the reasons I liked the previous implementation is that if the containment didn't change then neither did the eTag. But this was more consistent across DBs. Perhaps we just need to stream all containment and generate a hash in code rather than trying to do it in the database.